File: mpi/code/c/solution/helloworld_mpi.c
#include <stdlib.h>
#include <stdio.h>
#include <mpi.h>

int main(int argc, char* argv[])
{
  int nbTask;
  int myRank;

  MPI_Init(&argc, &argv);
  MPI_Comm_size(MPI_COMM_WORLD, &nbTask);
  MPI_Comm_rank(MPI_COMM_WORLD, &myRank);

  printf("I am task %d out of %d\n",myRank,nbTask);

  MPI_Finalize();
  return 0;
}

File: mpi/code/fortran/sujet/helloworld_mpi.f90
program helloworld_mpi

  use mpi

  implicit none

  integer   :: nbTask, myRank, ierr

  call MPI_Init(ierr)

  call MPI_COMM_SIZE(MPI_COMM_WORLD, nbTask, ierr)
  call MPI_COMM_RANK(MPI_COMM_WORLD, myRank, ierr)

  write (*,*) 'I am task', myRank, 'out of',nbTask

  call MPI_Finalize(ierr)

end program helloworld_mpi
